Quarkus does not start/respond when running inside an Azure Function on Java 11

I've generated a Quarkus Azure Functions project using the provided Maven archetype (following this guide: https://quarkus.io/guides/azure-functions-http). Everything is stock in the generated project, except I've updated the Azure Functions Maven plugin to 1.9.0 and set the following properties:
```
<java.version>11</java.version>
<maven.compiler.parameters>true</maven.compiler.parameters>
<maven.compiler.source>${java.version}</maven.compiler.source>
<maven.compiler.target>${java.version}</maven.compiler.target>
```
In addition, I've added a `runtime` block to the configuration of Azure Functions Maven plugin:
```
<runtime>
  <os>linux</os>
  <javaVersion>11</javaVersion>
</runtime>
```
As recommended by Microsoft, I've installed a Zulu build of OpenJDK, specifically "Zulu: 11.41.23" (I think that's based on OpenJDK 11.0.8+10), and set `JAVA_HOME` to the location of this JDK.

When executing `mvn install azure-functions:run`, Azure Functions starts OK and locates the function. I can access `locahost:7071` just fine and observe that the Azure Functions runtime environment is up and running. However, when trying to access `localhost:7071/api/hello` (which is an endpoint that Quarkus should respond to), I never get a response. The same behaviour can be observed if Azure Functions is started using the `func` tool. Adding the `--verbose` flag shows the request coming in, but never returning.

**Expected behavior**
I would expect to Quarkus to return immediatly with "hello jaxrs" or at least with an error message if something gone awry during startup or a 404 if the endpoint does not exist.

**To Reproduce**
Checkout https://github.com/cx-lmp/azure-functins-quarkus-java-11 for configuration and source code. As mentioned, it's basically generated by the Maven archetype with a minimal set of changes. 
Build the project and run Azure Functions with:
```
mvn install azure-functions:run
```
Visit `localhost:7071/api/hello` and observe that a response is never returned.

**Environment (please complete the following information):**
 - OS: Windows 10 version 2004
 - Terminal: Powershell Core 7.0.3
 - Quarkus version: 1.8.1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Maven 3.6.3
 - Azure Functions Core Tools (3.0.2912 Commit hash: bfcbbe48ed6fdacdf9b309261ecc8093df3b83f2)
 - Function Runtime Version: 3.0.14287.0

Hope I've provided you with enough information. Thanks!

I dived a bit deeper and found that there is an error at the initialization of the ApplicationImpl
```
[2020-11-08T08:18:40.648Z] java.lang.ExceptionInInitializerError
[2020-11-08T08:18:40.648Z] 	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:57)
[2020-11-08T08:18:40.648Z] 	at java.base/java.lang.Class.forName0(Native Method)
[2020-11-08T08:18:40.648Z] 	at java.base/java.lang.Class.forName(Class.java:315)
[2020-11-08T08:18:40.648Z] 	at eu.wecan.quarkus.MyFunction.<clinit>(MyFunction.java:31)
[2020-11-08T08:18:40.648Z] 	at java.base/java.lang.Class.forName0(Native Method)
For detailed output, run func with --verbose flag.
[2020-11-08T08:18:40.648Z] 	at java.base/java.lang.Class.forName(Class.java:398)
[2020-11-08T08:18:40.648Z] 	at com.microsoft.azure.functions.worker.broker.EnhancedJavaMethodExecutorImpl.getContainingClass(EnhancedJavaMethodExecutorImpl.java:63)
[2020-11-08T08:18:40.648Z] 	at com.microsoft.azure.functions.worker.broker.EnhancedJavaMethodExecutorImpl.<init>(EnhancedJavaMethodExecutorImpl.java:22)
[2020-11-08T08:18:40.648Z] 	at com.microsoft.azure.functions.worker.broker.FactoryJavaMethodExecutor.getJavaMethodExecutor(FactoryJavaMethodExecutor.java:20)
[2020-11-08T08:18:40.648Z] 	at com.microsoft.azure.functions.worker.broker.JavaFunctionBroker.loadMethod(JavaFunctionBroker.java:38)
[2020-11-08T08:18:40.648Z] 	at com.microsoft.azure.functions.worker.handler.FunctionLoadRequestHandler.execute(FunctionLoadRequestHandler.java:27)
[2020-11-08T08:18:40.648Z] 	at com.microsoft.azure.functions.worker.handler.FunctionLoadRequestHandler.execute(FunctionLoadRequestHandler.java:9)
[2020-11-08T08:18:40.649Z] 	at com.microsoft.azure.functions.worker.handler.MessageHandler.handle(MessageHandler.java:45)
[2020-11-08T08:18:40.649Z] 	at com.microsoft.azure.functions.worker.JavaWorkerClient$StreamingMessagePeer.lambda$onNext$0(JavaWorkerClient.java:92)
[2020-11-08T08:18:40.649Z] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[2020-11-08T08:18:40.649Z] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2020-11-08T08:18:40.649Z] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2020-11-08T08:18:40.649Z] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2020-11-08T08:18:40.649Z] 	at java.base/java.lang.Thread.run(Thread.java:834)
[2020-11-08T08:18:40.649Z] Caused by: java.lang.IllegalArgumentException: SRCFG00013: No Converter registered for class io.quarkus.runtime.configuration.MemorySize
[2020-11-08T08:18:40.649Z] 	at io.smallrye.config.SmallRyeConfig.lambda$getConverter$2(SmallRyeConfig.java:289)
[2020-11-08T08:18:40.649Z] 	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1737)
[2020-11-08T08:18:40.649Z] 	at io.smallrye.config.SmallRyeConfig.getConverter(SmallRyeConfig.java:286)
[2020-11-08T08:18:40.649Z] 	at io.quarkus.runtime.generated.Config.<clinit>(Config.zig:415)
[2020-11-08T08:18:40.649Z] 	... 19 more
```
It is called from the static initializer of `io.quarkus.azure.functions.resteasy.runtime.BaseFunction`
https://github.com/quarkusio/quarkus/blob/02e0ed5772f471ae901996f27e705b203e2d4d94/extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/BaseFunction.java#L47

I'm a bit stuck with the debugging, since I'm just learning quarkus since yesterday and have no idea how to access these .zig generated sources.

> Caused by: java.lang.IllegalArgumentException: SRCFG00013: No Converter registered for class io.quarkus.runtime.configuration.MemorySize

Are you sure all your artifacts are consistent and you don't have old artifacts around?

It looks a lot like a problem you would have with mixed versions.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -149,7 +149,7 @@
         <aws-xray.version>2.4.0</aws-xray.version>
         <awssdk.version>2.15.77</awssdk.version>
         <aws-alexa-sdk.version>2.37.1</aws-alexa-sdk.version>
-        <azure-functions-java-library.version>1.3.0</azure-functions-java-library.version>
+        <azure-functions-java-library.version>1.4.0</azure-functions-java-library.version>
         <kotlin.version>1.4.20</kotlin.version>
         <dekorate.version>0.14.2</dekorate.version>
         <maven-artifact-transfer.version>0.10.0</maven-artifact-transfer.version>

--- a/devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/singleton-examples/azure-functions-http-example/base/azure-config/function.tpl.qute.json
+++ b/devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/singleton-examples/azure-functions-http-example/base/azure-config/function.tpl.qute.json
@@ -1,5 +1,5 @@
 {
-  "scriptFile" : "../{project.artifact-id}-{project.version}-runner.jar",
+  "scriptFile" : "../{project.artifact-id}-{project.version}.jar",
   "entryPoint" : "io.quarkus.azure.functions.resteasy.runtime.Function.run",
   "bindings" : [ {
     "type" : "httpTrigger",

--- a/devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/singleton-examples/azure-functions-http-example/base/pom.tpl.qute.xml
+++ b/devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/singleton-examples/azure-functions-http-example/base/pom.tpl.qute.xml
@@ -1,10 +1,10 @@
 <project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
   <properties>
-    <azure.functions.maven.plugin.version>1.3.2</azure.functions.maven.plugin.version>
+    <azure.functions.maven.plugin.version>1.9.1</azure.functions.maven.plugin.version>
     <resources-plugin.version>3.1.0</resources-plugin.version>
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
     <functionAppName>$\{artifactId}-{gen-info.time}</functionAppName>
     <functionAppRegion>{app-region}</functionAppRegion>
     <functionResourceGroup>{resource-group}</functionResourceGroup>
@@ -23,21 +23,22 @@
           <resourceGroup>$\{functionResourceGroup}</resourceGroup>
           <appName>$\{functionAppName}</appName>
           <region>$\{functionAppRegion}</region>
+          <runtime>
+            <!-- runtime os, could be windows, linux or docker-->
+            <os>windows</os>
+            <javaVersion>11</javaVersion>
+            <!-- for docker function, please set the following parameters -->
+            <!-- <image>[hub-user/]repo-name[:tag]</image> -->
+            <!-- <serverId></serverId> -->
+            <!-- <registryUrl></registryUrl>  -->
+          </runtime>
           <appSettings>
             <!-- Run Azure Function from package file by default -->
             <property>
-              <name>WEBSITE_RUN_FROM_PACKAGE</name>
-              <value>1</value>
-            </property>
-            <property>
               <name>FUNCTIONS_EXTENSION_VERSION</name>
-              <value>~2</value>
+              <value>~3</value>
             </property>
-            <property>
-              <name>FUNCTIONS_WORKER_RUNTIME</name>
-              <value>java</value>
-            </property>
-          </appSettings>
+           </appSettings>
         </configuration>
       </plugin>
       <plugin>
@@ -106,6 +107,25 @@
                   </includes>
                 </resource>
               </resources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <!-- azure plugin wants a specific name for the jar so need to copy runner jar and rename it -->
+      <plugin>
+        <groupId>com.coderplus.maven.plugins</groupId>
+        <artifactId>copy-rename-maven-plugin</artifactId>
+        <version>1.0.1</version>
+        <executions>
+          <execution>
+            <id>rename-file</id>
+            <phase>package</phase>
+            <goals>
+              <goal>copy</goal>
+            </goals>
+            <configuration>
+              <sourceFile>$\{project.build.directory}/$\{project.artifactId}-$\{project.version}-runner.jar</sourceFile>
+              <destinationFile>$\{stagingDirectory}/$\{project.artifactId}-$\{project.version}.jar</destinationFile>
             </configuration>
           </execution>
         </executions>

--- a/docs/src/main/asciidoc/azure-functions-http.adoc
+++ b/docs/src/main/asciidoc/azure-functions-http.adoc
@@ -15,6 +15,8 @@ One azure function deployment can represent any number of JAX-RS, servlet, Vert.
 
 include::./status-include.adoc[]
 
+NOTE: Only text based media types are supported at the moment as Azure Functions HTTP Trigger for Java does not support a binary format
+
 == Prerequisites
 
 To complete this guide, you need:

--- a/extensions/azure-functions-http/maven-archetype/src/main/resources/archetype-resources/azure-config/function.json
+++ b/extensions/azure-functions-http/maven-archetype/src/main/resources/archetype-resources/azure-config/function.json
@@ -1,5 +1,5 @@
 {
-  "scriptFile" : "../${project.build.finalName}-runner.jar",
+  "scriptFile" : "../${project.build.finalName}.jar",
   "entryPoint" : "io.quarkus.azure.functions.resteasy.runtime.Function.run",
   "bindings" : [ {
     "type" : "httpTrigger",

--- a/extensions/azure-functions-http/maven-archetype/src/main/resources/archetype-resources/pom.xml
+++ b/extensions/azure-functions-http/maven-archetype/src/main/resources/archetype-resources/pom.xml
@@ -8,8 +8,8 @@
   <properties>
     <compiler-plugin.version>3.8.1</compiler-plugin.version>
     <maven.compiler.parameters>true</maven.compiler.parameters>
-    <maven.compiler.target>1.8</maven.compiler.target>
-    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <maven.compiler.source>11</maven.compiler.source>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <quarkus-plugin.version>999-SNAPSHOT</quarkus-plugin.version>
@@ -19,7 +19,7 @@
     <resources-plugin.version>3.1.0</resources-plugin.version>
     <surefire-plugin.version>3.0.0-M5</surefire-plugin.version>
 
-    <azure.functions.maven.plugin.version>1.3.2</azure.functions.maven.plugin.version>
+    <azure.functions.maven.plugin.version>1.9.1</azure.functions.maven.plugin.version>
     <functionAppName>${appName}</functionAppName>
     <functionAppRegion>${appRegion}</functionAppRegion>
     <functionResourceGroup>${resourceGroup}</functionResourceGroup>
@@ -112,19 +112,19 @@
           <resourceGroup>${functionResourceGroup}</resourceGroup>
           <appName>${functionAppName}</appName>
           <region>${functionAppRegion}</region>
+          <runtime>
+            <!-- runtime os, could be windows, linux or docker-->
+            <os>windows</os>
+            <javaVersion>11</javaVersion>
+            <!-- for docker function, please set the following parameters -->
+            <!-- <image>[hub-user/]repo-name[:tag]</image> -->
+            <!-- <serverId></serverId> -->
+            <!-- <registryUrl></registryUrl>  -->
+          </runtime>
           <appSettings>
-            <!-- Run Azure Function from package file by default -->
-            <property>
-              <name>WEBSITE_RUN_FROM_PACKAGE</name>
-              <value>1</value>
-            </property>
             <property>
               <name>FUNCTIONS_EXTENSION_VERSION</name>
-              <value>~2</value>
-            </property>
-            <property>
-              <name>FUNCTIONS_WORKER_RUNTIME</name>
-              <value>java</value>
+              <value>~3</value>
             </property>
           </appSettings>
         </configuration>
@@ -177,24 +177,23 @@
               </resources>
             </configuration>
           </execution>
-          <!-- copy the uber-jar into Azure staging directory -->
+        </executions>
+      </plugin>
+      <!-- azure plugin wants a specific name for the jar so need to copy runner jar and rename it -->
+      <plugin>
+        <groupId>com.coderplus.maven.plugins</groupId>
+        <artifactId>copy-rename-maven-plugin</artifactId>
+        <version>1.0.1</version>
+        <executions>
           <execution>
-            <id>copy-uberjar</id>
-            <phase>install</phase>
+            <id>rename-file</id>
+            <phase>package</phase>
             <goals>
-              <goal>copy-resources</goal>
+              <goal>copy</goal>
             </goals>
             <configuration>
-              <overwrite>true</overwrite>
-              <outputDirectory>${stagingDirectory}</outputDirectory>
-              <resources>
-                <resource>
-                  <directory>${project.build.directory}</directory>
-                  <includes>
-                    <include>${project.artifactId}-${project.version}-runner.jar</include>
-                  </includes>
-                </resource>
-              </resources>
+              <sourceFile>${project.build.directory}/${project.artifactId}-${project.version}-runner.jar</sourceFile>
+              <destinationFile>${stagingDirectory}/${project.artifactId}-${project.version}.jar</destinationFile>
             </configuration>
           </execution>
         </executions>

--- a/extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/BaseFunction.java
+++ b/extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/BaseFunction.java
@@ -34,12 +34,13 @@ import io.quarkus.vertx.http.runtime.VertxHttpRecorder;
 public class BaseFunction {
     private static final Logger log = Logger.getLogger("io.quarkus.azure");
 
-    protected static final String deploymentStatus;
+    protected static String deploymentStatus;
     protected static boolean started = false;
+    protected static boolean bootstrapError = false;
 
     private static final int BUFFER_SIZE = 8096;
 
-    static {
+    protected static void initQuarkus() {
         StringWriter error = new StringWriter();
         PrintWriter errorWriter = new PrintWriter(error, true);
         if (Application.currentApplication() == null) { // were we already bootstrapped?  Needed for mock azure unit testing.
@@ -50,7 +51,8 @@ public class BaseFunction {
                 app.start(args);
                 errorWriter.println("Quarkus bootstrapped successfully.");
                 started = true;
-            } catch (Exception ex) {
+            } catch (Throwable ex) {
+                bootstrapError = true;
                 errorWriter.println("Quarkus bootstrap failed.");
                 ex.printStackTrace(errorWriter);
             }

--- a/extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/Function.java
+++ b/extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/Function.java
@@ -14,7 +14,10 @@ public class Function extends BaseFunction {
     public HttpResponseMessage run(
             @HttpTrigger(name = "req") HttpRequestMessage<Optional<String>> request,
             final ExecutionContext context) {
-        if (!started) {
+        if (!started && !bootstrapError) {
+            initQuarkus();
+        }
+        if (bootstrapError) {
             HttpResponseMessage.Builder responseBuilder = request
                     .createResponseBuilder(HttpStatus.valueOf(500)).body(
                             deploymentStatus.getBytes(StandardCharsets.UTF_8));

